src/Messenger.Client/Services/Impl/MessengerSenderActionRequest.cs

```diff
@@ -0,0 +1,14 @@
+﻿using Messenger.Client.Objects;
+using Newtonsoft.Json;
+
+
+namespace Messenger.Client.Services.Impl
+{
+    internal class MessengerSenderActionRequest
```

Comment:
It should be part Messenger.Client.Objects to follow the current structure and renamed to something like MessengerSenderAction

Reply:
That's actually the first approach I went for, but reason I changed to enum is that for the sender actions call there is no other associated data passed through beyond which action you want to perform, thus a simple enum is both simplest to achieve this and cleanest from a consumption perspective (no wider wrapper object needed).

The other problem if we were to expose a full model object is that one of biggest pain points of the current library is there is no mechanism to map enums in the model objects, and they are all passed through as raw strings instead. If we were to make the param a full object it would end up having a single property and that property would be a raw string – ie: the wrong abstraction for the consumer of the SDK. I would actually love to see all the Messenger.Client.Objects to be switched to an enum-based pattern instead of all raw strings – but I think that would be a much bigger refactoring that should be applied to entire library separately (and perhaps with a major version rev).


In the meantime, keeping it as an internal implementation detail through your existing Messenger.Client.Services.Impl objects (and the newly introduced 'MessengerSenderActionRequest' one) seems like the best way to go. This hides the detail from consumption-side while allowing us to switch over to proper enum mappings in the future without changing the SDK interface for this particular call. Do note that the MessengerSenderAction enum is indeed in the Messenger.Client.Objects namespace, so I think this does use both the right external interface as well as internal one to achieve the goals at hand.

Reply:
Hmm I just noticed the 'MessengerXxxxType' objects which hold the possible values for enumerated members.

Was the reason you went for this approach (raw strings + const string values) as opposed to C# enums so that it can be forward-compatible with any new members that get introduced prior to the library being updated to account for them?

It unfortunately would make the SDK consumption even more awkward since it would mean exposing the entire MessengerMessage object and the consumer having to know that they need to set either the 'Text/Attachment/QuickReplies' properties OR the 'SenderAction' one but not both. Curious to your thoughts on this one....